src/test/addrman_tests.cpp

```diff
@@ -816,19 +816,21 @@ BOOST_AUTO_TEST_CASE(addrman_selecttriedcollision)
     for (unsigned int i = 1; i < 23; i++) {
         CService addr = ResolveService("250.1.1." + ToString(i));
         BOOST_CHECK(addrman.Add({CAddress(addr, NODE_NONE)}, source));
-        addrman.Good(addr);
 
-        // No collisions yet.
-        BOOST_CHECK(addrman.size() == i);
+        // No collisions in tried.
+        BOOST_CHECK(addrman.Good(addr));
```

Comment:
This test seems very limited, because the output is always `"[::]:0"`.
But since reviewing this change requires revisiting this logic, maybe might as well improve it to test broader behavior?

Reply:
I agree, this test is very limited. Ideally, it would:

1. create a collision and test that `SelectTriedCollision()` returns a value
2. resolve collisions
2. try to enter a duplicate and test that `SelectTriedCollision()` does not return a value

I didn't make those changes because I was trying to keep the scope of the PR limited to just refactoring the usage of `Good()`, but perhaps you're right and it's better to improve the test while refactoring? I can update the PR description to be a bit more general

Reply:
Yeah, your intention would usually make sense, but here I found myself digging into the addrman logic while reviewing.

I could be easier on review i guess :)

Reply:
The more complicated scenarios are covered in the subsequent tests `addrman_noevict` and `addrman_evictionworks`. I think that's fine - maybe just the naming of `addrman_selecttriedcollision` is not ideal because it suggests it covers all about selecttriedcollision that there is to test.

Reply:
@mzumsande yeah i agree, it's more of a naming problem